inspirehep/modules/records/cli.py

```diff
@@ -112,6 +112,65 @@ def next_batch(iterator, batch_size):
     return batch
 
 
+def get_query_records_to_index(pid_types):
+    """
+    Return a query for retrieving all non deleted records by pid_type
+
+    Args:
+        pid_types(List[str]): a list of pid types
+
+    Return:
+        SQLAlchemy query for non deleted record with pid type in `pid_types`
+    """
+    query = (
+        db.session.query(PersistentIdentifier.object_uuid).join(RecordMetadata, type_coerce(PersistentIdentifier.object_uuid, String) == type_coerce(RecordMetadata.id, String))
```

Comment:
why we need ``db.session.query`` and not ``Model.query``?

Reply:
Well, that's how the SQLAlchemy documents the `join` https://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.join
The `query` object is not callable, thus you can not do something like `Model.query(Model.id).join(...`